src/index.rs

```diff
@@ -529,36 +608,29 @@ impl ResourceIndex {
                 "Any collision must involve at least 2 resources"
             );
             *collisions -= 1;
-
-            if *collisions == 1 {
+            if *collisions <= 1 {
```

Comment:
Shouldn't `collisions == 0` be treated as a broken invariant?

Reply:
Yes 
Reverted this change